tools/profiles/extensions/minimal-printf.json

```diff
@@ -1,12 +1,12 @@
 {
     "GCC_ARM": {
-       "common": ["-DMBED_MINIMAL_PRINTF", "-fno-builtin-printf"],
+       "common": ["-DMBED_MINIMAL_PRINTF"],
```

Comment:
Why was this removed?

Reply:
Because the change is making printf work "normally", so the compiler no longer needs to treat it as "abnormal".

Without this option, the compiler can transform `printf("Hello\n")` into `puts("Hello")`, or `printf(".")` into `putchar('.')`. That would have triggered the problem we're now fixing.

Reply:
Ok. I need to understand how that retarget work.

Reply:
It's not so much about how the retargetting works in detail as the fact you're short-circuiting it

`printf`, `putchar` and `puts` are all supposed to go to the same place as `fputc(stdout)`.

If those things don't all go to the same place, or skip the buffering among only some paths, then things get weird when you mix them.

(The retargetting controls where fputc(stdout) goes, and then all other I/O calls follow).